go/client/logclient.go

```diff
@@ -158,28 +158,22 @@ func (c *LogClient) addChainWithRetry(ctx context.Context, ctype ct.LogEntryType
 }
```

Comment:
(floating comment, github won't let me attach it in the right place!)
Now that we're passing ctx everywhere we should never be given a nil context.
So you should remove the 
```go
if ctx == nil {
  ctx = context.TODO()
}
```
lines in `addChainWithRetry` above.